Classes/SegmentService.swift

```diff
@@ -8,6 +8,9 @@
 
 import Foundation
 
+public typealias SegmentResult<T: SegmentBatchCodable> = Result<[T], Error>
+public typealias DefaultSegmentResult = SegmentResult<BatchElement>
```

Comment:
added these typealiases as a convenience. one side-effect of this change is having to declare the type in the closure (shown in the example image). not really a big deal but this way you can do `(result: SegmentResult<MyType>) in` instead of `(result: Result<[MyType], Error>) in`
and the `BatchElement` remains as a simple default implementation

Reply:
So this isn't a major issue from a technical perspective, but I think we should change the default BatchElement implementation to DefaultBatchElement. The Segment SDK will always (unless they do a complete overhaul of their implementation strategy) have an array of calls, labeled 'batch'. So any T that we use will still be a BatchElement, just with slightly different parameters.

Reply:
makes sense. can change that and also the subtypes for properties and traits and whatnot if you agree that makes sense as well